### PR TITLE
Feature/gp 31 forgot passwod page

### PR DESCRIPTION
**Pull request by TobbyToe**
1. The user through forgot password page send an email to the user's email.
2. When it is successful, the user will receive a success message, else, the user will get an error status code.
3. The user clicks the email link (URL with token) will link to the reset password page
4. The reset password page use can enter the password and confirm the password, then submit the form to the backend.
5. When it is successful, the user will link to Success Page with a back to login button, else Link to Fail Page.

**[x]** email API in forgot password page use a **`put`** method.
**[x]** reset API, get token by `windows.location.search` and `.substr(1)` aim to get token without question mark. then put token into the API and send back to backend.
**[x]** pages with scss (ForgotPasswordPage, ResetPasswordPage, ResetPasswordSuccessPage, ResetPasswordFailPage )
**[x]** ForgotPasswordPage use a simple return the success message by `?success(XXX) : ()` 
**[x]** Clone the backend to test whether API works for the backend, and it works with **`put`** method.
**[x]** Leave for interaction, split regex to meet Joi error messages.

### CODE REVIEW

File: src/pages/ResetPasswordPage/ResetPasswordPage.jsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        .catch((err) => {
+          setIsLoading(false);
+          setFormError(err.response.data);
+        });
+    } catch (err) {
+      setFormError(err);
+    }
+  };
+
+  useEffect(() => {}, [success]);
```
[the remark this answers is not in the document]
Comment (TobbyToe) [reply src/pages/ResetPasswordPage/ResetPasswordPage.jsx:74]: 本地 测试的时候用的 忘了删除。

File: src/pages/ForgotPasswordPage/ForgotPasswordPage.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      label: 'Email:',
+      id: 'email',
+      placeholder: 'Your Email Address',
+      state: email,
+      setState: setEmail,
+      schema: Joi.string().email()
+    }
+  ];
+
+  const FP_API_URL = `http://localhost:3001/api/v1/passwordReset`;
```
[the remark this answers is not in the document]
Comment (TobbyToe) [reply src/pages/ForgotPasswordPage/ForgotPasswordPage.jsx:24]: `https://service.greatphermesia.mianora.top/api/v1/passwordReset`

[the remark this answers is not in the document]
Comment (TobbyToe) [reply src/pages/ForgotPasswordPage/ForgotPasswordPage.jsx:24]: as for reset part
`https://service.greatphermesia.mianora.top/api/v1/passwordReset/${getUrlToken}`

File: src/pages/ForgotPasswordPage/ForgotPasswordPage.jsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  const handleForgetPasswordFormSubmission = (event) => {
+    event.preventDefault();
+    setIsLoading(true);
+    setFormError('');
+
+    try {
```
[the remark this answers is not in the document]
Comment (TobbyToe) [reply src/pages/ForgotPasswordPage/ForgotPasswordPage.jsx:30]: delete try catch

File: src/pages/ResetPasswordPage/ResetPasswordPage.scss [hunk 4]
```diff
@@ -0,0 +1,46 @@
+@use '../../sass/mixin';
+
+.reset_password_page {
+  @include mixin.useFormContainer();
+  &--content {
+    color: #000;
+  }
+}
+.reset_password_page {
+  display: flex;
```
[the remark this answers is not in the document]
Comment (TobbyToe) [reply src/pages/ResetPasswordPage/ResetPasswordPage.scss:10]: merged in one

File: src/pages/ForgotPasswordPage/ForgotPasswordPage.jsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      label: 'Email:',
+      id: 'email',
+      placeholder: 'Your Email Address',
+      state: email,
+      setState: setEmail,
+      schema: Joi.string().email()
+    }
+  ];
+
+  const FP_API_URL = `https://service.greatphermesia.mianora.top/api/v1/passwordReset`;
```
Comment (Ollie-lee) [inline src/pages/ForgotPasswordPage/ForgotPasswordPage.jsx:24]: 你们hostname是有环境变量的吧？我看见有其他同学写过，你global search下，不要像这样写死

**Comment by Ollie-lee** [comment]
conficts

File: src/pages/ForgotPasswordPage/ForgotPasswordPage.jsx (hunk 5 above)
Comment (NorrisWu0) [reply src/pages/ForgotPasswordPage/ForgotPasswordPage.jsx:24]: 有的，用的是`REACT_APP_API_URL`作为base，变量具体位置在`PROJECTROOT/.env`.

File: src/pages/ForgotPasswordPage/ForgotPasswordPage.jsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    event.preventDefault();
+    setIsLoading(true);
+    setFormError('');
+
+    try {
+      axios
+        .put(FP_API_URL, { email })
+        .then((response) => {
+          setIsLoading(false);
+          if (response.status === 200) setSuccess(true);
```
[the remark this answers is not in the document]
Comment (TobbyToe) [reply src/pages/ForgotPasswordPage/ForgotPasswordPage.jsx]: Ternary expression of IF
`response.status === 200
          ? setSuccess(true)
          : setFormError(response.body);`
with eslintrc.json add`"no-unused-expressions": [2, { "allowTernary": true }]`

File: src/pages/ResetPasswordPage/ResetPasswordPage.jsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+        })
+        .catch((err) => {
+          setIsLoading(false);
+          setFormError(err.response.data);
+        });
+    } catch (err) {
+      setFormError(err);
+    }
+  };
+
```
[the remark this answers is not in the document]
Comment (TobbyToe) [reply src/pages/ResetPasswordPage/ResetPasswordPage.jsx:69]: delete `try` `.catch` change if into Ternary Expression by 
`response.status === 200
          ? setSuccess(true)
          : setFormError(response.body);`
with eslintrc added `"no-unused-expressions": [2, { "allowTernary": true }]`

**Review by TobbyToe — COMMENTED** [review]
changes as required, 
[x] Updated the `**API**` into `**variable URL**`;
[x] Updated `**if ternary expression**`, code with `**eslintrc.json**`;
[x] Delete useless codes (it was used by localhost testing);
[x] Solved scss two same className conflictions, merge in one;

**Review by Ollie-lee — APPROVED** [review]

**Pull request merged**